### PR TITLE
Drop kairos and use sdk for collector

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

**Review by mauromorales — APPROVED** [review]
lgtm

**Review by jimmykarily — APPROVED** [review]
There is a broken test but that's also broken in `main`. Looks good.

File: cmd/discovery/client/config.go [hunk 1]
```diff
@@ -28,12 +27,14 @@ type Config struct {
 func unmarshalConfig() (Config, error) {
 	var result Config
 
-	o := &collector.Options{NoLogs: true}
+	o := &collector.Options{NoLogs: true, MergeBootCMDLine: false}
```
Comment (jimmykarily) [inline cmd/discovery/client/config.go:30]: This isn't changing anything right? We never used cmdline for kcrypt-challenger. We are just being explicit now?

Comment (Itxaka) [reply cmd/discovery/client/config.go:30]: Yep. Cmdline should only be applicable to cloud config stuff like the agent. 

In any case, this should default to false on te struct but I wanted to be very explicit as this now requires a filter function and I'm passing an empty one that does nothing, and that is only triggered if the cmdline is parsed

**Comment by Itxaka** [comment]
gonna manually test as Im not sure about the local encryption. IIRC it should print a message after the install saying that its encrypting something but I cant see anything in the logs mentioning this.... I hope its not broken already sigh...

**Comment by Itxaka** [comment]
> gonna manually test as Im not sure about the local encryption. IIRC it should print a message after the install saying that its encrypting something but I cant see anything in the logs mentioning this.... I hope its not broken already sigh...

locally works as expected.

One thing to note is that now the challenger iso is being built against ubuntu 23 so maybe something has changed there that affects CI.

**Pull request merged**